### PR TITLE
feat: Changed pdf datetime to event timezones

### LINKED ISSUE #7438: Change email and PDF datetimes to event timezone
They are currently shown in server timezone

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #7438 

#### Short description of what this resolves:
feat: Changed pdf datetime to event timezones

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by codedsun** [comment]
@iamareebjamal - Review

File: app/models/event_invoice.py [hunk 1]
```diff
@@ -86,6 +87,14 @@ def previous_month_date(self):
     def due_at(self):
         return self.issued_at + timedelta(days=EventInvoice.DUE_DATE_DAYS)
 
+    @property
+    def issued_at_tz(self):
+        return self.issued_at.astimezone(pytz.timezone(self.event.timezone))
+
+    @property
+    def due_at_tz(self):
```
Comment (iamareebjamal) [inline app/models/event_invoice.py:95]: Please use datetime helper, it has timezone support already

File: app/templates/pdf/ticket_attendee.html [hunk 2]
```diff
@@ -61,8 +61,8 @@
             {% endif %}
 
             <span class="label">Date and Time</span><br>
-            <span style="font-size: 15px;">From: {{ order.event.starts_at | datetime(order.event.timezone) }}</span><br>
-            <span style="font-size: 15px;">To: {{ order.event.ends_at | datetime(order.event.timezone) }}</span>
+            <span style="font-size: 15px;">From: {{ order.event.starts_at_tz | datetime(order.event.timezone) }}</span><br>
+            <span style="font-size: 15px;">To: {{ order.event.ends_at_tz | datetime(order.event.timezone) }}</span>
```
Comment (iamareebjamal) [inline app/templates/pdf/ticket_attendee.html:65]: This was already according to event timezone

File: app/models/event_invoice.py (hunk 1 above)
Comment (codedsun) [reply app/models/event_invoice.py:95]: @iamareebjamal `return self.due_at.astimezone(datetime.astimezone(self.event.timezone))` like this?

Comment (codedsun) [reply app/models/event_invoice.py:95]: Or you meant `datetime(order.event.timezone)` like this in jinja template?

**Comment by codedsun** [comment]
@iamareebjamal - Review

File: app/models/order.py [hunk 3]
```diff
@@ -145,6 +146,14 @@ def ticket_pdf_path(self) -> str:
             + '.pdf'
         )
 
+    @property
+    def created_at_tz(self):
+        return self.created_at.astimezone(pytz.timezone(self.event.timezone))
+
+    @property
+    def completed_at_tz(self):
```
Comment (iamareebjamal) [inline app/models/order.py:154]: As I said in last review, not needed

File: app/templates/pdf/attendees_pdf.html [hunk 4]
```diff
@@ -97,10 +97,10 @@ <h3 style="text-align:center;">{{ holders[0].event.name }} - {{ ("Attendees List
 
                     {% if holder.order and holder.order.status == 'completed' %}
                         <span
-                            class="datetime">{{ holder.order.completed_at | datetime }} - {{ holder.order.completed_at | humanize }}</span>
+                            class="datetime">{{ holder.order.completed_at_tz | datetime }} - {{ holder.order.completed_at_tz | humanize }}</span>
```
Comment (iamareebjamal) [inline app/templates/pdf/attendees_pdf.html:100]: Use the datetime helper as suggested in last review

File: app/templates/pdf/event_invoice.html [hunk 5]
```diff
@@ -17,11 +17,11 @@
...
           position: relative;
-          margin: 0 auto; 
+          margin: 0 auto;
           color: #001028;
-          background: #FFFFFF; 
-          font-family: Arial, sans-serif; 
-          font-size: 12px; 
+          background: #FFFFFF;
+          font-family: Arial, sans-serif;
+          font-size: 12px;
```
Comment (iamareebjamal) [inline app/templates/pdf/event_invoice.html:24]: Why?

File: app/templates/pdf/orders.html [hunk 6]
```diff
@@ -103,9 +103,9 @@ <h3 style="text-align:center;">{{ event.name }} - {{ ("Order List") }}</h3>
                                 {{ ('NA') }}
                             {% endif %}<br>
                             {% if order.status == 'completed' %}
-                                <span class="datetime">{{ order.completed_at | datetime }}</span>
+                                <span class="datetime">{{ order.completed_at_tz | datetime }}</span>
```
Comment (iamareebjamal) [inline app/templates/pdf/orders.html:106]: Not needed

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Moved even further from expected since last review

File: app/templates/pdf/event_invoice.html (hunk 5 above)
Comment (codedsun) [reply app/templates/pdf/event_invoice.html:24]: Its done during the commit..commit hooks format this file

Comment (codedsun) [reply app/templates/pdf/event_invoice.html:24]: @iamareebjamal

File: app/templates/pdf/event_invoice.html [hunk 7]
```diff
@@ -167,8 +167,8 @@ <h1>INVOICE {{ invoice.identifier }}</h1>
         {% if user.billing_tax_info %}
           <div><span>TAX ID</span> {{user.billing_tax_info or ''}}</div>
         {% endif %}
-        <div><span>DATE</span> {{ invoice.issued_at | date }}</div>
-        <div><span>DUE DATE</span> {{ invoice.due_at | date }}</div>
+        <div><span>DATE</span> {{ invoice.issued_at | datetime(event.timezone) }}</div>
+        <div><span>DUE DATE</span> {{ invoice.due_at | datetime(event.timezone) }}</div>
```
Comment (iamareebjamal) [inline app/templates/pdf/event_invoice.html:171]: This is date and not datetime, make date helper timezone aware

File: app/templates/pdf/event_invoice.html [hunk 8]
```diff
@@ -167,8 +167,8 @@ <h1>INVOICE {{ invoice.identifier }}</h1>
         {% if user.billing_tax_info %}
           <div><span>TAX ID</span> {{user.billing_tax_info or ''}}</div>
         {% endif %}
-        <div><span>DATE</span> {{ invoice.issued_at | date }}</div>
-        <div><span>DUE DATE</span> {{ invoice.due_at | date }}</div>
+        <div><span>DATE</span> {{ invoice.issued_at | date(event.timezone) }}</div>
+        <div><span>DUE DATE</span> {{ invoice.due_at | date(event.timezone) }}</div>
```
Comment (iamareebjamal) [inline app/templates/pdf/event_invoice.html:171]: The date filter is not timezone aware. It'll fail. Please test the changes before asking for review. Reduce the request review cycle please. The changes on this PR either go opposite of what it should be or plain don't work.

Why don't you test it first?

Comment (codedsun) [reply app/templates/pdf/event_invoice.html:171]: I didn't requested for review this tym. I will test and then ask for review and will make sure to reduce the request review cycle. Sorry for the inconvinience

File: app/templates/flask_ext/jinja/filters.py [hunk 9]
```diff
@@ -30,10 +30,10 @@ def simple_datetime_display(date, timezone='UTC', format='%B %d, %Y %I:%M %p'):
         return date.astimezone(pytz.timezone(timezone)).strftime(format)
 
     @app.template_filter('date')
-    def simple_date_display(date):
+    def simple_date_display(date, timezone='UTC'):
         if not date:
             return ''
-        return date.strftime('%B %d, %Y')
+        return date.astimezone(pytz.timezone(timezone)).strftime('%B %d, %Y')
```
Comment (iamareebjamal) [inline app/templates/flask_ext/jinja/filters.py:36]: return simple_datetime_display(date, timezone, '%B %d, %Y')

**Comment by codedsun** [comment]
@iamareebjamal - Review

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/open-event-server/pull/7452#discussion_r525305843

File: app/templates/flask_ext/jinja/filters.py [hunk 10]
```diff
@@ -30,10 +30,10 @@ def simple_datetime_display(date, timezone='UTC', format='%B %d, %Y %I:%M %p'):
         return date.astimezone(pytz.timezone(timezone)).strftime(format)
 
     @app.template_filter('date')
-    def simple_date_display(date):
+    def simple_date_display(date, timezone='UTC'):
         if not date:
             return ''
```
Comment (iamareebjamal) [inline app/templates/flask_ext/jinja/filters.py:35]:
```suggestion
```

**Comment by iamareebjamal** [comment]
Other emails contain time as well

**Comment by codedsun** [comment]
https://github.com/fossasia/open-event-server/blob/247e9bf29f8f1ba6ed762bcb1952bedb39205366/app/api/helpers/system_mails.py#L277

https://github.com/fossasia/open-event-server/blob/247e9bf29f8f1ba6ed762bcb1952bedb39205366/app/api/helpers/system_mails.py#L292

https://github.com/fossasia/open-event-server/blob/247e9bf29f8f1ba6ed762bcb1952bedb39205366/app/api/helpers/system_mails.py#L307

https://github.com/fossasia/open-event-server/blob/247e9bf29f8f1ba6ed762bcb1952bedb39205366/app/api/helpers/system_mails.py#L323

@iamareebjamal Correct?

**Comment by iamareebjamal** [comment]
After event email and others as well

**Comment by codedsun** [comment]
@iamareebjamal As you mentioned after event email

https://github.com/fossasia/open-event-server/blob/247e9bf29f8f1ba6ed762bcb1952bedb39205366/app/api/helpers/system_mails.py#L144-L152

Are you referring to sent at field?

**Comment by iamareebjamal** [comment]
No, I must have forgot. If that's it, then yes, those are the only emails

**Comment by codedsun** [comment]
@iamareebjamal - Tested

```
subject : Your invoice for Event Till March for Oct 2020 is available on Open Event

message:
Hello Suneet  Srivastava,<br><br>The invoice for your event Event Till March on eventyay.com for Oct 2020 is now available. Your invoice payment is due within 30 days.<br><br>Amount Due: INR 100.0<br><br>Please pay within 30 days.<br><br>Pay Now at: <a href='http://eventyay.com/event-invoice/22345ad1/review'>http://eventyay.com/event-invoice/22345ad1/review</a><br><br>A detailed invoice is available in <a href='https://eventyay.com/account/billing/invoices/'>the billing area</a> of your account. If you have any questions about invoices, please find more information on our FAQ at https://support.eventyay.com.<br><br><em>Thank you for using Open Event!</em><br><br>Open Event Team

```

image

**Comment by codedsun** [comment]
Thanks! yeah 🎉

**Pull request merged**